tests/tensor_search/integ_tests/test_dict_score_modifiers.py

```diff
@@ -0,0 +1,573 @@
+                    )
+                )
+                # Search with score modifier
+                # 0.5 * 0.5 * 4 = 1 (1 and 7)
+                score_modifier = ScoreModifier(**{
+                    "add_to_score": [{"field_name": "map_score_mods_long.a", "weight": 20},
+                                     {"field_name": "score_mods_long", "weight": 20}],
+                }
+                                               )
+                res = tensor_search.search(
+                    index_name=index.name, config=self.config, text="",
+                    score_modifiers=score_modifier,
+                    result_count=10
+                )
+
+                # Get the score of the first result.
+                score_of_first_result = res["hits"][0]["_score"]
+                score_of_second_result = res["hits"][1]["_score"]
+
+                # anticipated score is 4294967295012*20 = 85899345900240
+                # Assert that the first and second result both have _ids in 1 and 2
+                # and 85899345900239 <= score <= 85899345900241
+                self.assertTrue(res["hits"][0]["_id"] in ["2", "1"])
+                self.assertTrue(res["hits"][1]["_id"] in ["2", "1"])
+                self.assertTrue(85899345900239 <= score_of_first_result <= 85899345900241)
+                self.assertTrue(85899345900239 <= score_of_second_result <= 85899345900241)
+
+    def test_unstructured_unsupported_map_error(self):
+        """
+        Test that only the document errors out, not the whole batch
+        """
+        for index in [self.unstructured_default_text_index]:
+            with self.subTest(index=index.type):
+                # Add documents
+                res = tensor_search.add_documents(
+                    config=self.config,
+                    add_docs_params=AddDocsParams(
+                        index_name=index.name,
+                        docs=[
+                            {"_id": "1", "text_field": "a photo of a cat", "map_score_mods_long": {"a": 4294967295012}},
+                            {"_id": "2", "text_field": "a photo of a cat", "score_mods_long": 4294967295012},
+                            {"_id": "3", "text_field": "a photo of a cat", "score_mods_long": 1,
+                             "unsupported_map": {"text": "hello", "number": 42}},
+                            {"_id": "4", "text_field": "a photo of a cat", "score_mods_long": 1,
+                             "supported_number_map": {"number": 42}},
+                            {"_id": "5", "text_field": "a photo of a cat", "map_score_mods_int": {"c": 1},
+                             "map_score_mods_float": {"a": 0.5}},
+                        ],
+                        tensor_fields=["text_field"] if isinstance(index, UnstructuredMarqoIndex) else None,
+                    )
+                )
+
+                # Expected
+                expected = {"1": 200, "2": 200, "3": 400, "4": 200, "5": 200}
```

Comment:
Verify the message. You could be getting 400 because of something else

Reply:
done